### PR TITLE
test/librbd: clean up unused TEST_COOKIE variable

### PR DESCRIPTION
**Pull request by Svelar**
Clean it up because
```
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/librbd/exclusive_lock/test_mock_PreAcquireRequest.cc:47:26: error: unused variable 'TEST_COOKIE' [-Werror,-Wunused-const-variable]
static const std::string TEST_COOKIE("auto 123");
                         ^
1 error generated.
```

Ref: https://jenkins.ceph.com/job/ceph-pull-requests-arm64/59242/consoleFull#-5120798632a811ea2-3e7b-466b-84b4-d13df7e35809

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by Svelar** [comment]
jenkins test make check arm64

**Comment by Svelar** [comment]
jenkins test make check

File: src/test/librbd/exclusive_lock/test_mock_PreAcquireRequest.cc [hunk 1]
```diff
@@ -44,8 +44,6 @@ using ::testing::SetArgPointee;
 using ::testing::StrEq;
 using ::testing::WithArg;
 
-static const std::string TEST_COOKIE("auto 123");
```
Comment (idryomov) [inline src/test/librbd/exclusive_lock/test_mock_PreAcquireRequest.cc:47]: A similar variable seems to be unused in `src/test/librbd/exclusive_lock/test_mock_PostAcquireRequest.cc` and `src/test/librbd/exclusive_lock/test_mock_PreReleaseRequest.cc`.  Is there a reason to clean up only one instance?

Comment (Svelar) [reply src/test/librbd/exclusive_lock/test_mock_PreAcquireRequest.cc:47]: My initial intention was to make as mirror revise as possible. But yes, I would clean up all instances as you said, and then ask for your review again~

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
@Svelar This showed up in reef and is failing regular (not arm64) `make check` there.  I will do the backports, but I'd like to know what changed -- any idea?

**Comment by Svelar** [comment]
> @Svelar This showed up in reef and is failing regular (not arm64) `make check` there. I will do the backports, but I'd like to know what changed -- any idea? These variables have been unused for years...

Weird right? Because it happened in omani001, I find the libgcc version which clang++ links is different with other servers. But Iam not sure the root cause.

**Comment by idryomov** [comment]
> Seems like `__cpp_lib_is_constant_evaluated` is default defined because of clang++ or g++ version?
> So unused variable error appears.
> Ref: https://github.com/ceph/fmt/blob/a33701196adfad74917046096bf5a2aa0ab0bb50/include/fmt/core.h#L340-L353

@Svelar Are you suggesting that a helper in fmt submodule (in its "detail" namespace) affects the entire build?  Even if so, that wouldn't explain it popping up randomly.

I think it's more likely that the environment on some of the Jenkins builders gets messed up from time to time.

**Comment by Svelar** [comment]
> > Seems like `__cpp_lib_is_constant_evaluated` is default defined because of clang++ or g++ version?
> > So unused variable error appears.
> > Ref: https://github.com/ceph/fmt/blob/a33701196adfad74917046096bf5a2aa0ab0bb50/include/fmt/core.h#L340-L353
> 
> @Svelar Are you suggesting that a helper in fmt submodule (in its "detail" namespace) affects the entire build? Even if so, that wouldn't explain it popping up randomly.
> 
> I think it's more likely that the environment on some of the Jenkins builders gets messed up from time to time.

Actually it is not randomly, versions of libgcc differs in servers. Not sure why some are libgcc12, some are libgcc11.

**Pull request merged**